data-prepper-plugin-framework/src/main/java/org/opensearch/dataprepper/plugin/VariableExpander.java

```
@@ -57,7 +57,36 @@ public <T> T translate(final JsonParser jsonParser, final Class<T> destinationTy
 
                     })
                     .findFirst()
-                    .orElseGet(() -> objectMapper.convertValue(rawValue, destinationType));
+                    .orElseGet(() -> {
+                        // This change is to support any call
+                        // to validate the secret with a placeholder secret expression like "AWS_SECRET_EXPRESSION"
+                        // which is not of a secrets expression that we would check (filter check above fails) but
+                        // still expects an instance of PluginConfigVariable object returned
+                        if (destinationType.equals(PluginConfigVariable.class)) {
+                            return destinationType.cast(new PluginConfigVariable() {
+                                @Override
+                                public Object getValue() {
+                                    return rawValue;
+                                }
+
+                                @Override
+                                public void setValue(Object updatedValue) {
+
```

Should we throw `new UnsupportedOperationException()`?

---

Yes. Added it.